### ISSUE TITLE
Compatibility with Sails

### ISSUE THREAD

**Issue by Kikobeats** [auth0/express-jwt#52]
I love this repository and I'm trying to use it in a Sails project.

Express middleware in sails as known as [policies](http://sailsjs.org/#/documentation/concepts/Policies) and internally sails use express 3.x.

Simply you can define a policies and decide what route request pass for each policies (if is necessary).

The interface for create the policy is the same to create a middleware: 

```js
module.exports = function(req, res, next) {
 // do something
};
```
I want to validate the header token with `express-jwt` and later do a custom action in the policy  (for example, search the user and expose in `req.user`) I'm not sure how to concatenate the middlewares to get in the more external middleware the callback of the `express-jwt`.

Just something like this:

```js
var options = {secret: 'seeeecret!'};
var expressjwt = require('express-jwt')({options});

module.exports = function(req, res, next) {
  expressjwt(req, res, function(token){
    // Now I have the token
    // do something
    next();
  });
};
```

Do you think that is possible?

**Comment by jfromaniello** [comment]
Sorry, I missunderstand your comment in the gist. You can do that:

```javascript
var options = {secret: 'seeeecret!'};
var expressjwt = require('express-jwt')({options});

module.exports = function(req, res, next) {
  expressjwt(req, res, function(token){
    getUser(req.user.id, function (err, user) {
      if (err) return next(err);
      req.user = user;
      next()
    });
  });
};
```

**Comment by Kikobeats** [comment]
thanks!! Only have a little problem:

```
{
"name": "JsonWebTokenError",
"message": "invalid signature"
}
```

I'm using standard settings :S

**Comment by jfromaniello** [comment]
Yes, feel free to send a pull request

**Comment by Kikobeats** [comment]
@jfromaniello man with the new update, the code is broken. I detect that is for [this line in index.js](https://github.com/auth0/express-jwt/blob/master/lib/index.js#L78) because the token is a string, not a json. I don't know how to fix :(

**Comment by jfromaniello** [comment]
Yes, it looks like we broke it with this commit:

https://github.com/auth0/express-jwt/commit/672dd72b5e2132a5947220a24539fbbb58ee105a

@dschenkelman ?

**Comment by Kikobeats** [comment]
I can not determine a use case where you intend to use a JSON... all time you are passing a token string, right?

anyway, check first if token is json/object before put the flag can be a safe mode.

**Comment by jfromaniello** [comment]
> I can not determine a use case where you intend to use a JSON... all time you are passing a token string, right?

no, you can create JWTs where the payload is a JSON, like this:

```
var jwt = require('jsonwebtoken');
var token = jwt.sign({foo: 'bar'}, secret);
```

in fact most of the time we use this instead of string.

I've fixed this on version `v1.3.1`.

Please next time create another issue, this issue has been closed already.

Thanks for reporting it
